### ISSUE TITLE
[Bug] Delete backups after upload toggle attempting to delete all from Google Drive

### ISSUE THREAD

**Issue by caraar12345** [sabeechen/hassio-google-drive-backup#745]
###### Description:
Truly a fantastic add on - but just found an interesting bug. When turning on "delete backups after upload", it's meant to just ignore the "how many to keep on Home Assistant", but it's **also** ignoring the "how many to keep on Google Drive". And when that's set to `0` (don't delete anything), it's actually interpreting it as `0` (save nothing). So it just tried to delete all my backups! 

_Thanks for the "no more than 1 deletion" stopper...:laughing:_

---

 Addon version: `0.109.1`
 Home Assistant Version: `2022.11.2`
 Supervisor Version: `2022.10.2`
 Supervisor Channel: `beta`
 Hassos Version: `9.3`
 Docker Version: `20.10.18`
 Architecture: `amd64`
 Machine: `generic-x86-64`
 Date: `2022-11-13T13:31:58.168047+00:00`
 Timezone: `Europe/London`
 Failure Time: `2022-11-13T13:31:10.037163+00:00`
 Last Good Sync: `2022-11-13T13:31:58.045438+00:00`
 ###### Exception:
 ```
 No error could be identified automatically.
 ```
 Backups:
 ```
 {
    "HomeAssistant": {
        "backups": 1,
        "retained": 0,
        "deletable": 1,
        "name": "HomeAssistant",
        "title": "Home Assistant",
        "latest": "2022-11-13T13:31:41Z",
        "max": 10,
        "enabled": true,
        "icon": "home-assistant",
        "ignored": 32,
        "detail": "",
        "size": "0.0 B",
        "ignored_size": "2.5 GB",
        "free_space": "17.8 GB"
    },
    "GoogleDrive": {
        "backups": 271,
        "retained": 0,
        "deletable": 271,
        "name": "GoogleDrive",
        "title": "Google Drive",
        "latest": "2022-11-10T13:10:54Z",
        "max": 9999999,
        "enabled": true,
        "icon": "google-drive",
        "ignored": 0,
        "detail": "aaron@aaroncarson.co.uk",
        "size": "223.1 GB",
        "ignored_size": "0.0 B"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_BACKUPS_IN_HA": 10,
    "Setting.MAX_BACKUPS_IN_GOOGLE_DRIVE": 9999999,
    "Setting.DAYS_BETWEEN_BACKUPS": 1.0,
    "Setting.DELETE_AFTER_UPLOAD": true,
    "Setting.IGNORE_IPV6_ADDRESSES": true
}
 ```

**Comment by sabeechen** [comment]
Yikes, this is a very legitimate bug.  This isn't the first time hat "Multiple Delete" safety check has saved me a lot of embarrassment and my users a lot of pain.  It could be even worse though, if you have these settings and delete your last backup the addon will just loop around infinitely creating and deleting backups.  Whoops!

The fix is simple thankfully, I'll release a new version with a patch.  I'm kind of surprised no one has reported this before you. 

Side note, this is a very good bug report.  You've:
- Described the problem plainly
- Included your configuration 
- Found the workaround yourself (setting "Max Backups in Google Drive" very high).

It might seem pedantic of me to point this out, but I get so, so, so many awful bug reports and solid ones like this just make it much easier to identify and fix the problem.

**Comment by sabeechen** [comment]
This should now be fixed in the latest version, [0.109.2](https://github.com/sabeechen/hassio-google-drive-backup/releases/tag/v0.109.2).
